Index: src/main/webapp/resources/js/datatablesUtil.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/webapp/resources/js/datatablesUtil.js	(date 1535988360000)
+++ src/main/webapp/resources/js/datatablesUtil.js	(date 1535988395595)
@@ -20,18 +20,12 @@
     $("#modalTitle").html(i18n["editTitle"]);
     $.get(ajaxUrl + id, function (data) {
         $.each(data, function (key, value) {
-            form.find("input[name='" + key + "']").val(
-                key === "dateTime" ? formatDate(value) : value
-            );
+            form.find("input[name='" + key + "']").val(value);
         });
         $('#editRow').modal();
     });
 }
 
-function formatDate(date) {
-    return date.replace('T', ' ').substr(0, 16);
-}
-
 function deleteRow(id) {
     $.ajax({
         url: ajaxUrl + id,
@@ -79,8 +73,10 @@
 
 function failNoty(jqXHR) {
     closeNoty();
+    // https://stackoverflow.com/questions/48229776
+    var responseJSON = JSON.parse(jqXHR.responseText);
     failedNote = new Noty({
-        text: "<span class='fa fa-lg fa-exclamation-circle'></span> &nbsp;" + i18n["common.errorStatus"] + ": " + jqXHR.status + (jqXHR.responseJSON ? "<br>" + jqXHR.responseJSON : ""),
+        text: "<span class='fa fa-lg fa-exclamation-circle'></span> &nbsp;" + i18n["common.errorStatus"] + ": " + jqXHR.status + (responseJSON ? "<br>" + responseJSON : ""),
         type: "error",
         layout: "bottomRight"
     }).show();
Index: src/main/webapp/resources/js/mealDatatables.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/webapp/resources/js/mealDatatables.js	(date 1535988360000)
+++ src/main/webapp/resources/js/mealDatatables.js	(date 1535988395608)
@@ -14,6 +14,19 @@
     $.get(ajaxUrl, updateTableByData);
 }
 
+// http://api.jquery.com/jQuery.ajax/#using-converters
+$.ajaxSetup({
+    converters: {
+        "text json": function (stringData) {
+            var json = JSON.parse(stringData);
+            $(json).each(function () {
+                this.dateTime = this.dateTime.replace('T', ' ').substr(0, 16);
+            });
+            return json;
+        }
+    }
+});
+
 $(function () {
     datatableApi = $("#datatable").DataTable({
         "ajax": {
@@ -24,13 +37,7 @@
         "info": true,
         "columns": [
             {
-                "data": "dateTime",
-                "render": function (date, type, row) {
-                    if (type === 'display') {
-                        return formatDate(date);
-                    }
-                    return date;
-                }
+                "data": "dateTime"
             },
             {
                 "data": "description"
